### PR TITLE
Feature/vtk refactoring

### PR DESCRIPTION
**Pull request by martinfalk**
Changes:
* major refactoring regarding TTK, VTK, and tensorvis modules
* the updated vtk module can now be used on its own without relying on TTK 
    + the previous `misc/vtk` module was replaced with the VTK parts extracted from the `topovis/ttk` module
    + refactored Python script in `misc/vtk/tools/` for generating separate processores based on ParaView VTK and TTK filters
* the TensorVisIO module now uses the updated vtk module
* added a VTK overlay port to address problem with multiple `find_package(VTK)` calls, see https://github.com/microsoft/vcpkg/issues/35223

### CODE REVIEW

File: misc/vtk/readme.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+This module depends on VTK, it is therefore recommended to use vcpkg for managing and building VTK.
+Use a CMake preset similar to this to enable this VTK module:
+
+```json
+    {
+      "name": "vtk-msvc-vcpkg",
+      "displayName": "VTK MSVC VCPKG",
+      "inherits" : ["modules", "msvc-dev-vcpkg"],
+      "cacheVariables": {
+        "VCPKG_TARGET_TRIPLET" :   "x64-windows",
```
Comment (petersteneteg) [inline misc/vtk/readme.md:14]: not relevant

File: misc/vtk/tools/vtkwrapping/fixes.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    className: str
+    prop_identifier: str
+
+
+class TextReplacement(NamedTuple):
+    old_value: str
+    new_value: str
+
+
+vtk_fixes: dict[PropertyKey, TextReplacement] = {}
```
Comment (petersteneteg) [inline misc/vtk/tools/vtkwrapping/fixes.py:43]: don't use global variables

File: tools/vcpkg/vtk/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,394 @@
+set(VTK_SHORT_VERSION 9.2)
```
Comment (petersteneteg) [inline tools/vcpkg/vtk/portfile.cmake:1]: Explain what you have changed compared the upstream port

File: topovis/ttk/readme.md [hunk 4]
```diff
@@ -1,60 +1,28 @@
...
-        "Path" : "C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.31.31103/bin/Hostx64/x64;$penv{Path}"
-      },
       "cacheVariables": {
-        "IVW_CFG_VCPKG_OVERLAYS" : "${sourceDir}/tools/vcpkg;${sourceParentDir}/modules/tools/vcpkg",
-        "IVW_MODULE_TTK" :          { "type": "BOOL", "value": "ON" }
-      }
-    }
+        "VCPKG_TARGET_TRIPLET" :   "x64-windows",
+        "VCPKG_FEATURE_FLAGS":     "-compilertracking",
+        "VCPKG_INSTALL_OPTIONS":   "--x-abi-tools-use-exact-versions",
```
Comment (petersteneteg) [inline topovis/ttk/readme.md:15]: unrelated remove

File: topovis/ttk/readme.md [hunk 5]
```diff
@@ -1,60 +1,28 @@
...
-Generate wrappers around all the ttkvtk filters by parsing the paraview xml files
-describing the filters.
+This module provides processors wrapping the [Topology ToolKit](https://topology-tool-kit.github.io/), short TTK. The processors are based on the XML information on the various TTK ParaView filters.
 
-to build with vcpkg add  ```"topology-tool-kit"```
-to the vcpkg.json file in the main inviwo repo
-that will install vtk and ttk.
-
-For setting stuff up in cmake is use this preset:
+The ttk module depends on the Inviwo VTK module (`misc/vtk`) as well as the TTK and VTK libraries. The VTK functionality is provided by the Inviwo module. To simplify things, we recommend the usage of vcpkg for both TTK and VTK using the following CMake preset.
```
Comment (petersteneteg) [inline topovis/ttk/readme.md:5]: This is windows only

Comment (petersteneteg) [reply topovis/ttk/readme.md:5]: might explain how to adjust for other systems

**Review by petersteneteg — CHANGES_REQUESTED** [review]

File: misc/vtk/tools/vtkwrapping/config.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class Mode(Enum):
+    vtk = 1
+    ttk = 2
+    custom = 3
+
+
+@dataclass
+class Config:
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply misc/vtk/tools/vtkwrapping/config.py:42]: This is more explicit than just using argparse and contains additional settings like the path to clang-format. Also, it is recommended not to use Enum in combination with argparse (`Mode`).

File: misc/vtk/tools/vtkwrapping/commandlineargs.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        print(f'{parser.prog} error: mode set to "ttk" requires the following argument: --ttkrepo',
+              file=sys.stderr)
+        parser.exit()
+    if args.mode == "custom" and not args.filters:
+        parser.print_usage()
+        print(f'{parser.prog} error: mode set to "custom" requires the following argument: --filters',  # noqa: E501
+              file=sys.stderr)
+        parser.exit()
+
+    config.global_config.destination = args.output
```
[the remark this answers is not in the document]
Comment (martinfalk) [reply misc/vtk/tools/vtkwrapping/commandlineargs.py:78]: This is more explicit than just using argparse and contains additional settings like the path to clang-format. Also, it is recommended not to use Enum in combination with argparse (Mode).

File: topovis/ttk/readme.md (hunk 5 above)
Comment (martinfalk) [reply topovis/ttk/readme.md:5]: Which would be to use the installed packages?

Comment (petersteneteg) [reply topovis/ttk/readme.md]: Think I have used vcpkg on all platforms for vtk

**Review by petersteneteg — APPROVED** [review]

**Pull request merged**